### ISSUE TITLE
Fetch interrupting x3dom rendering

### ISSUE THREAD

**Issue by oli1230** [jamesleesaunders/d3-x3dom#74]
I am using Javascript's fetch function to get live data from a server which somehow interferes with x3dom's ability to render in the webpage. The x3d, scene, and all shape components are present when examined in the debugger, but important CSS and event-based functionalities are not being implemented properly.

**Comment by oli1230** [comment]
[graph.zip](https://github.com/jamesleesaunders/d3-x3dom/files/3179728/graph.zip)

**Comment by jamesleesaunders** [comment]
Hi Oli,

It look a bit of head scratching but I have managed to get your example going.

Here are the changes I made to your file:
https://github.com/jamesleesaunders/d3-x3dom/commit/5256bdf1051dbdbfca60c57e7e3493873f10aa32

and working example here:
https://raw.githack.com/jamesleesaunders/d3-x3dom/oli-stein/examples/finalGraph.html

Key changes:
* Removed your <div> tags from the header (you shouldn't use div tags in the HTML header).
* Added x3dom.reload(); which refreshed the x3dom after the async had completed.
* Slightly improved the data manipulation (although I am sure this could be improved/simplified further with map or filter).
* Loaded the chart in the .then() (after data loaded).
